microsoft/PowerToys
# [FancyZones] App zone history on multiple desktops

## Summary of the Pull Request
1. Add support to save zone history over multiple desktops for single app.
2. After virtual desktop is deleted, remove app zone history as well.


## PR Checklist
* [x] Applies to #2684 and #2678


## Validation Steps Performed
Multiple desktops app zone history:
1. Create more than one desktop and assign some layout to them.
2. Assign two instances of the same application to different desktops.
3. Close those instances.
4. Re-open them again.
5. Check `C:\Users\<user>\AppData\Local\Microsoft\PowerToys\FancyZones` files `app-zone-history.json`.

Expected result: Information about zone history of single application contains information for different desktops, after application is opened, it goes to the different zone based on which desktop it is.

App zone history data of deleted desktop:
1. Create more than one desktop and assign some layout to them.
2. Assign some applications into zones on all desktops.
3. Delete one of them.
4. Check `C:\Users\<user>\AppData\Local\Microsoft\PowerToys\FancyZones` files `app-zone-history.json` and `zones-settings.json`.

Expected result: No data belonging to deleted desktops is found.
